### PR TITLE
Rails 5.2+ only

### PR DESCRIPTION
**Pull request by danielricecodes**
This PR offers a new version of the gem that is limited to Rails 5.2+ only.  This greatly simplified the code in places and removed all the inner classes from `UniquenessWithoutDeletedValidator`.

### CODE REVIEW

**Comment by danielricecodes** [comment]
@mvz - This is for your review I believe.  Here's a version of ActsAsParanoid for only Rails 5.2 and higher.

**Comment by mvz** [comment]
Thanks, @danielricecodes. This looks good at first glance. I'll have a more thorough look later in the week.

**Comment by mvz** [comment]
I'm going to first release a version from current master because it contains some fixes that pre-5.2 users may be interested in. After that, I'll merge this and release a 5.2+ only version.

File: .travis.yml [hunk 1]
```diff
@@ -5,19 +5,13 @@ cache:
   - directories:
     - vendor/bundle
 rvm:
-  - 2.2.10
-  - 2.3.8
-  - 2.4.5
```
Comment (mvz) [inline .travis.yml]: I'd like to keep 2.4.5 since that is supported by Rails 5.2.

**Comment by BrianHawley** [comment]
@mvz when this gets merged, make sure that there continues to be a branch that we can target for fixing issues when still using rails 4.2. Even after something like 0.7.0 comes out with these changes, we'll need to be able to have future 0.6.x releases, for projects that still need to use Rails 4.2 for now.

**Comment by mvz** [comment]
@BrianHawley I'll prepare a branch for that.

**Comment by mvz** [comment]
@danielricecodes can you please rebase this on current master?

**Comment by danielricecodes** [comment]
> @danielricecodes can you please rebase this on current master?

If the build passes do you still want me to rebase?  I'd have to force push to undo the merge commit.

File: .travis.yml (hunk 1 above)
Comment (danielricecodes) [reply .travis.yml]: I put them back and configured the travis configuration file to only test these rubies with Rails 5.2.

**Comment by danielricecodes** [comment]
@mvz - rebased and ready I believe. :+1:

**Comment by mvz** [comment]
Thanks, @danielricecodes!

File: .travis.yml (hunk 1 above)
Comment (mvz) [reply .travis.yml]: Oh wow I didn't realize Rails 5.2 actually supports Ruby 2.2.

**Pull request merged**